Servers/UI/OJS.Servers.Ui/ClientApp/src/redux/services/contestsService.ts

```diff
@@ -62,6 +71,47 @@ export const contestsService = createApi({
                         strategy,
                     },
                 }),
+                keepUnusedDataFor: 0,
+            }),
+        getContestUserParticipation: builder.query<ICompeteContestResponseType, { id: number; isOfficial: boolean }>({
+            query: ({ id, isOfficial }) => ({
+                url: `/compete/${id}`,
+                params: { isOfficial },
+            }),
+            keepUnusedDataFor: 0,
+        }),
+        submitContestSolution: builder.mutation<void, ISubmitContestSolutionParams>({
+            query: ({ content, official, problemId, submissionTypeId }) => ({
+                url: '/Compete/Submit',
+                method: 'POST',
+                body: { content, official, problemId, submissionTypeId },
+            }),
+        }),
+        submitContestSolutionFile: builder.mutation<void, ISubmitContestSolutionParams>({
+            query: ({ content, official, submissionTypeId, problemId }) => ({
+                url: '/Compete/SubmitFileSubmission',
+                method: 'POST',
+                body: { content, official, problemId, submissionTypeId },
+            }),
+        }),
+        submitContestPassword: builder.mutation<void, ISubmitContestPasswordParams>({
+            query: ({ contestId, isOfficial, password }) => ({
+                url: `/contests/SubmitContestPassword/${contestId}`,
+                method: 'POST',
+                params: { isOfficial },
+                body: { password },
+            }),
+        }),
+        // this should replace: getContestRegisteredUser
+        registerUserForContest: builder.mutation<
```

Comment:
Isn't this replacing getContestRegisteredUser already?

Reply:
yes, forgot to remove the comment

Reply:
done